GeoLib/SurfaceGrid.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * \date 2012-09-22
+ * \brief Declaration of the SurfaceGrid class.
+ *
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#ifndef SURFACEGRID_H_
+#define SURFACEGRID_H_
+
+#include <array>
+#include <vector>
+
+#include <boost/optional.hpp>
+
+#include "AABB.h"
+
+namespace GeoLib {
+
+// forward declarations
+class Triangle;
+class Surface;
+
+class SurfaceGrid : public AABB<GeoLib::Point> {
+public:
+	friend GeoLib::Surface;
```

Comment:
move a friend declaration to private. conceptually public don't need to know it.